irqstat

```diff
@@ -95,72 +119,77 @@ def filter_found(name, filter_list):
 
 
 def display_itop(batch, seconds, rowcnt, iterations, sort, totals, dispnode,
-                 zero, filters):
+                 zero, filters, file1, file2, overall, numafile):
     """Main I/O loop"""
     irqs = {}
     cpunodes = {}
     numacores = {}
     loops = 0
     width = len('NODEXX')
 
-    if batch:
+    if not file1 and batch:
         print "Running in batch mode"
-    else:
+    elif not file1:
         print ("interactive commands -- "
-               "t: view totals, 0-9: view node, any other key: quit")
+               "t: view totals, 0-9: view node, any other key: quit\r")
+
+    if file1:
+        intr_filename=file1
```

Comment:
Yes, good point! This is definitely something I forgot to do before submitting.

However, does this comment mean that `pylint` should find something about the lines above? Because I don't see anything. I don't even see the entries you're mentioning, TBH.

What I've found (mostly in previous commits) are issues with `(` of some `print` statements I was adding...

I installed it from my distro, and this is what I got:
```
pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 20 2020, 17:41:41) [GCC]
```

Reply:
Here's what I get:
```C: 63, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:137, 0: Exactly one space required around assignment
        intr_filename=file1
                     ^ (bad-whitespace)
C:139, 0: Exactly one space required around assignment
        intr_filename='/proc/interrupts'
                     ^ (bad-whitespace)
C:147, 0: No space allowed before bracket
        with open (intr_filename, 'r') as intr_file:
                  ^ (bad-whitespace)
C:299, 0: Exactly one space required around assignment
            intr_filename=file2
                         ^ (bad-whitespace)
C:359, 0: Exactly one space required around assignment
        options.batch=True
                     ^ (bad-whitespace)
C:368, 0: Exactly one space required around assignment
        options.overall=True
                       ^ (bad-whitespace)
C: 51, 4: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
C: 66,12: Variable name "nf" doesn't conform to snake_case naming style (invalid-name)
C: 72,12: Variable name "a" doesn't conform to snake_case naming style (invalid-name)
C: 85,16: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
C: 87,16: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
W: 59,12: Unused variable 'status' (unused-variable)
C:180,28: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary)
C:203,28: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary)
R:209,16: Unnecessary "else" after "return" (no-else-return)
R:194, 8: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
C:250,15: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
R:308, 0: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

Looks like my version is older:
```$ pylint --version
pylint 1.9.5,
astroid 1.6.6
Python 2.7.16 (default, Jun  5 2020, 22:59:21)
```

Not everything reported by pylint is "important" but including a pylint.rc with the repo saves time in avoiding comments/discussions over style.  It does also catch bigger issues, and I've learned a lot about python from its findings.

Reply:
Ah, ok, thanks for showing your output. Those seem good remarks to me, I'm not sure why my (newer) version suppresses them.

Anyway, I think I've taken care of them. I will update the pull request.

Reply:
From pylint:
```
C: 62, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C: 63, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C: 88, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:131, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:224, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:226, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:227, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:246, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:282, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C:363, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
R:209,16: Unnecessary "else" after "return" (no-else-return)
```

Reply:
Ok, I fixed the last one (`R:209,16: Unnecessary "else" after "return" (no-else-return)`), and I'm updating my branch.

The `print` ones have to do with Python 3, I think? In fact, your `pylint` complains if `(` are there, mine complains if they're not there.

How do we want to settle this? :-)

Reply:
Python 3 wins :)

https://docs.python.org/3.0/whatsnew/3.0.html#print-is-a-function

The easiest solution is just to disable superfluous-parens in pylint.rc, so I've done that.